kubernetes-itests/src/test/java/io/fabric8/crd/bicycle/Bicycle.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.bicycle;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Version;
+
+@Version("v1")
+@Group("demos.fabric8.io")
+public class Bicycle extends CustomResource<BicycleSpec, BicycleStatus> {
+  private String apiVersion;
```

Comment:
I don't like this. There should be only one source of truth.

Reply:
I had to do this because I wasn't able to get both versions working. Since CustomResource.setApiVersion doesn't allow updating apiVersion, the payload sent always contained version configured via annotations. Due to this, I was getting this error:
```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://192.168.39.93:8443/apis/demos.fabric8.io/v1beta1/bicycles. Message: the API version in the data (demos.fabric8.io/v1) does not match the expected API version (demos.fabric8.io/v1beta1). Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=the API version in the data (demos.fabric8.io/v1) does not match the expected API version (demos.fabric8.io/v1beta1), metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=BadRequest, status=Failure, additionalProperties={}).
        at io.fabric8.kubernetes.
```

Reply:
I understand why you did it. My point, though, is that we need to find a better way to support this use case.